### PR TITLE
Build kiwix-lib only if meson is compiled with xapian.

### PR DESCRIPTION
**Pull request by mgautierfr**
Depends of openzim/libzim#465

### CODE REVIEW

**Comment by kelson42** [comment]
@mgautierfr Do we really want to do that now? Not convinced this is a good idea as there is no real need to do this now.

File: meson.build [hunk 1]
```diff
@@ -39,6 +38,12 @@ else
   error('Cannot found header mustache.hpp')
 endif
 
+libzim_dep = dependency('libzim', version : '>=6.3.0', static:static_deps)
+if not compiler.has_header_symbol('zim/zim.h', 'LIBZIM_WITH_XAPIAN')
```
Comment (veloman-yunkan) [inline meson.build:42]: Shouldn't `zim/zim_config.h` be checked instead?

**Comment by kelson42** [comment]
@mgautierfr Any feedback on this?

File: meson.build (hunk 1 above)
Comment (kelson42) [reply meson.build:42]: @mgautierfr ?

**Review by kelson42 — COMMENTED** [review]
Would be good to get that merged.

File: meson.build (hunk 1 above)
Comment (mgautierfr) [reply meson.build:42]: `zim.h` includes `zim_config.h`.
Both header file would work but I prefer to keep using `zim.h` as I tend to consider `zim_config.h` as a bit more internal.

This is really a matter of opinion here. There is no real separation (and documentation) of what can be directly including or not.

**Comment by kelson42** [comment]
@veloman-yunkan `zim.h` being the top header to be include, it sounds logicial to me to rely on it. Would that be ok to merge for you?

**Review by veloman-yunkan — APPROVED** [review]

**Pull request merged**